frontend/src/providers/QueryBuilder.tsx

```diff
@@ -40,37 +47,51 @@ export function QueryBuilderProvider({
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	const params = useParams();
 
-	// ** TODO: create state for queryBuilder
-	// ** TODO: create state for queryFormulas
+	const [queryBuilderData, setQueryBuilderData] = useState<QueryBuilderData>({
+		queryData: [],
+		queryFormulas: [],
+	});
 
-	// ** TODO: Find out the types of the queryBuilder state
-	// ** TODO: Check exist version of the functionality
-	const [queryBuilderData, setQueryBuilderData] = useState<unknown[]>([
-		{
-			queryData: '',
-			queryFormulas: '',
-		},
-	]);
+	// ** TODO: Also in the future need to add AddFormula and AddQuery and remove them.
 
 	const resetQueryBuilderData = useCallback((): void => {
 		setQueryBuilderData(initialQueryBuilderData);
 	}, []);
 
-	const handleSetQueryBuilderData = useCallback((): void => {}, []);
+	const handleSetQueryData = useCallback(
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		(index: number, queryData: IBuilderQuery): void => {},
+		[],
+	);
+	const handleSetFormulaData = useCallback(
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		(index: number, formulaData: IBuilderFormula): void => {},
+		[],
+	);
+
 	// ** TODO: Discuss with Palash how the state of the queryBuilder and queryFormulas
 	// ** TODO: should be filled from url
 
-	// ** TODO: create set function for state
-
 	// ** TODO: put these values and setter to the context value
 
+	useEffect(() => {
+		// ** TODO: add logic when page will change the query builder data clears
+		resetQueryBuilderData();
```

Comment:
would this be in return ()=>{
resetQueryBuilderData()
}

Reply:
I will update this pull request and remove useEffect from provider. I think it would be controllable when we will have ability to clear query builder data state whenever we want or by corresponding conditions instead of only resetting when page change.